be/src/exec/pipeline/scan/olap_scan_prepare_operator.cpp

```diff
@@ -38,6 +38,11 @@ bool OlapScanPrepareOperator::has_output() const {
     return !is_finished();
 }
 
+Status OlapScanPrepareOperator::set_finished(RuntimeState* state) {
+    _ctx->set_finished();
```

Comment:
`OlapScanOperator` is finished earlier than `OlapScanPrepareOperator` when short-circuit, so maybe we should call `_ctx->set_finished` in `OlapScanOperator::set_finished` not in `OlapScanPrepareOperator::set_finished`?

In general, if a `ExecNode` is composed to two operators `OpA` and `OpB` and `OpB` depends on `OpB`. Maybe we should call `ctx->set_finished` in `OpB` and `ctx->is_finiished` in `OpA`, because `OpB` is finished before `OpA` due to short-circuit.